app/helpers/preferences_helper.rb

```diff
@@ -0,0 +1,9 @@
+module PreferencesHelper
+  def uppercase_enabled?(user, organization)
+    uppercase_organization?(organization) || (user.preferences.uppercase? if user)
```

Comment:
This code may be a domain behavior, isn't it?

Reply:
Remember that user - organization pair forms a `Mumuki::Domain::Workspace`. Perhaps we could do `current_workspace.uppercase?`

Reply:
That way we also help potential preferences to be also course-dependent, since the whole idea behind workspaces is to generalize organization-user and course-user pairs.

Reply:
I agree with moving this to domain and/or workspace, but I don't get how that would help preferences be course-dependent (unless we start storing preferences in a workspace-like object instead of storing them in `User`, as it is right now).

Reply:
Making them course-dependant is out of scope now. It is just an option that gets enabled when using workspaces.

I am not even saying that preferences should be part of organization or course, but that some aspects of some of them may be affected by such domain objects. For example, you have declared: 


```
  def uppercase_organization?(organization)
    organization.kindergarten?
  end
```

In this example, uppercase mode is not organization dependant, but user-dependent as always. However, organization also has effect on this preference. The same could happen with some other courses or organization with future preferences. For example, a teacher may choose to force gamification, overriding the user setting. Again this is out of scope, but is the natural continuation of the stone you have already thrown :stuck_out_tongue:

Reply:
:+1: To go in this direction!